vb.go

```diff
@@ -0,0 +1,54 @@
+package sgbucket
+
+import (
+	"fmt"
+	"sync"
+)
+
+type VbucketNo uint32
+type VbucketSeq uint64
+
+// Per-vbucket sequence counter
+type VbucketSeqCounter interface {
+	Incr(vbNo VbucketNo) (VbucketSeq, error)
+	Get(vbNo VbucketNo) (VbucketSeq, error)
+}
+
+type mapVbucketSeqCounter struct {
+	data        map[VbucketNo]VbucketSeq
+	mu          *sync.Mutex
+	numVbuckets int
+}
+
+func NewMapVbucketSeqCounter(numVbuckets int) *mapVbucketSeqCounter {
+	return &mapVbucketSeqCounter{
+		data:        map[VbucketNo]VbucketSeq{},
+		mu:          &sync.Mutex{},
+		numVbuckets: numVbuckets,
+	}
+}
+
+func (msq mapVbucketSeqCounter) Incr(vbNo VbucketNo) (VbucketSeq, error) {
+	msq.mu.Lock()
+	defer msq.mu.Unlock()
+
+	if vbNo > VbucketNo(msq.numVbuckets-1) {
```

Comment:
Another potential drawback of this approach - for this review, it took me a 30 seconds to figure out that VbucketNo was just the named uint32, and not a helper function.